### PR TITLE
Issue264

### PR DESCRIPTION
**Pull request by mreboud**
Add a method to start the optimization of an analysis at a random point of the parameter space.
Closes issue #264

### CODE REVIEW

File: python/eos/analysis.py [hunk 1]
```diff
@@ -217,11 +217,15 @@ def optimize(self, start_point=None, **kwargs):
         """
         Optimize the log(posterior) and returns a best-fit-point summary.
 
-        :param start_point: Parameter point from which to start the optimization, with the elements in the same order as in eos.Analysis.varied_parameters. If not specified, optimization starts at the current parameter point.
+        :param start_point: Parameter point from which to start the optimization, with the elements in the same order as in eos.Analysis.varied_parameters.
+                            If set to "Random", optimization starts at the random point in the space of the priors.
```
Comment (dvandyk) [inline python/eos/analysis.py:221]: Can we use ``random`` (all lower case)?

File: python/eos/analysis.py [hunk 2]
```diff
@@ -217,11 +217,15 @@ def optimize(self, start_point=None, **kwargs):
...
+                            If not specified, optimization starts at the current parameter point.
         :param start_point: iterable, optional
         """
         if start_point == None:
             start_point = [float(p) for p in self.varied_parameters]
+        elif start_point == "Random":
```
Comment (dvandyk) [inline python/eos/analysis.py:227]: See above. This should now also be adopted for the ``eos-analysis`` script. I think that will require a little bit of modifcations, such that either a list or a string are accepted as the start point.

File: python/eos/analysis.py [hunk 3]
```diff
@@ -217,11 +217,15 @@ def optimize(self, start_point=None, **kwargs):
         """
```
Comment (dvandyk) [inline python/eos/analysis.py:217]: I was wondering if we should add a ``rng`` keyword argument here, such that for a random starting point we still have a deterministic behaviour.

**Review by dvandyk — COMMENTED** [review]
Left a few inline comments.

File: python/eos/analysis.py (hunk 3 above)
Comment (mreboud) [reply python/eos/analysis.py:217]: Definitely!

File: python/eos/analysis.py (hunk 2 above)
Comment (mreboud) [reply python/eos/analysis.py]: I added a new option, let me know if this looks ok

File: src/scripts/eos-analysis [hunk 4]
```diff
@@ -386,6 +392,20 @@ def cmd_find_mode(args):
...
+        _gof = eos.GoodnessOfFit(analysis.log_posterior)
+        _chi2 = _gof.total_chi_square()
+        if _chi2 < min_chi2:
+            gof = _gof
+            bfp = _bfp
+            min_chi2 = _chi2
+    elif args.seed:
+        seed = args.seed
+        rng = RNG(seed)
+        starting_point = [p.inverse_cdf(rng.uniform(0, 1)) for p in analysis.log_posterior.log_priors()]
```
Comment (dvandyk) [inline src/scripts/eos-analysis:405]: Isn't this duplicating code? I would have just called ``analysis.optimize("random", rng=rng)``

**Review by dvandyk — COMMENTED** [review]
Approved w/ exception of the call to ``eos.Analysis.optimize``, which seems to duplicate code.

File: src/scripts/eos-analysis (hunk 4 above)
Comment (mreboud) [reply src/scripts/eos-analysis:405]: You are correct, I don't know what I had in mind...

**Pull request merged**